[#44285] Update Toggl Documentation (#11369)
opf/openproject: docs/user-guide/time-and-costs/time-tracking/toggl-integration/README.md

keywords: time tracking, toggl integration
---

# Time tracking with Toggl Track

> To track time to a work package via a start and stop button, we offer an integration for the time recording software (or online timesheet software) Toggl Track ([https://toggl.com/track/](https://toggl.com/track/)).

Reviewer: @adam-op I think it is worth it to explain/clarify that the spent time gets saved in Toggl Track and not in OpenProject.

## Installation

You can install a browser add-in to track time via Toggl Track in OpenProject. This way, you can record spent times directly to a work package in OpenProject via a start/stop button. You need to register an account at Toggle in order to use this feature.

This integration can be added for example for Google Chrome as [Toggle Track extension](https://chrome.google.com/webstore/detail/toggl-button-productivity/oejgccbfbmkkpaidnkphaiaecficdnfn) and it also exists for Firefox.

## Configuration

On the cogwheel of the toggle track extension you reach the **Settings**, from there chose **Integrations**.

> For OpenProject Domains that are located on openproject.com you are able to chose **Openproject - openroject.com** from the list of default integrations.

Reviewer: ```suggestion
For OpenProject Domains that are located on openproject.com, which is the case for most Enterprise Cloud instances, you are able to chose **Openproject - openroject.com** from the list of default integrations.
```

> For any other OpenProject Domains you need to exactly add the **FQDN** to the **Custom URLs for Integration** and chose **OpenProject**. For example: community.openproject.org (wildcards are not supported by Toggle Track at the moment of writing)

Reviewer: I believe that the abbreviation FQDM is not a term that most people know. I learnt it just some weeks ago.

```suggestion
For any other OpenProject domains you need to exactly add the domain name (including the subdomain if present) to the **Custom URLs for Integration** and chose **OpenProject**. For example: community.openproject.org (wildcards are not supported by Toggle Track at the moment of writing)
```

![toggl-configuration-integrations-openproject](toggl-configuration-integrations-openproject.png)

> ## Tracking time from OpenProject to Toggle Track

Reviewer: ```suggestion
## Tracking time for OpenProject work packages to Toggl Track
```

> If this add-in is installed and the Integration URL is configured, a **Start/Stop Timer button** is displayed on the work package details view, which can be used to record times from OpenProject into toggl:

Reviewer: ```suggestion
If this add-in is installed and the _Integration URL_ is configured, a **Start/Stop Timer button** is displayed on the work package details view, which can be used to record times from OpenProject into Toggl:
```

![toggl-button-openproject-workpackage-detail-view](toggl-button-openproject-workpackage-detail-view.png)

> ## Using toggle tracked time in OpenProject

Reviewer: ```suggestion
## Copying time entries from Toggl Track to OpenProject
```

> If you would like to use the times tracked at toggle in OpenProject, you are able to transfer the recorded times back to OpenProject via the API if needed.

Reviewer: ```suggestion
If you would like to use the times tracked in Toggl Track in OpenProject and you are a programmer, you are able to transfer the recorded time entries back to OpenProject via the API.
```

Details can be found in our [API documentation](../../../../api/).

> Alternatively you could also take a look at this [simple open source ruby script](https://github.com/wielinde/toggl_track_to_openproject) which is not officially supported by OpenProject GmbH.

Reviewer: ```suggestion
Alternatively you could also take a look at this [simple open source ruby script](https://github.com/wielinde/toggl_track_to_openproject) which is not officially supported by OpenProject GmbH. It selects time entries from Toggl Track via the Toggl Track API, accumulates the entries by date and saves them in OpenProject via the OpenProject API. Please read the instructions carefully. If you don't understand the README, please don't use it.
```
